clash-prelude/src/Clash/Explicit/BlockRam.hs

```diff
@@ -247,62 +256,62 @@ replace it with the string @\"undefined\"@ in the first few leading outputs.
 
 Finally we get to using 'blockRam'. On FPGAs, 'Clash.Prelude.RAM.asyncRam' will
 be implemented in terms of LUTs, and therefore take up logic resources. FPGAs
-also have large(r) memory structures called /Block RAMs/, which are preferred,
+also have large(r) memory structures called /block RAMs/, which are preferred,
 especially as the memories we need for our application get bigger. The
-'blockRam' function will be translated to such a /Block RAM/.
+'blockRam' function will be translated to such a /block RAM/.
 
-One important aspect of Block RAMs have a /synchronous/ read port, meaning that,
-unlike the behavior of 'Clash.Prelude.RAM.asyncRam', given a read address @r@
-at time @t@, the value @v@ in the RAM at address @r@ is only available at time
-@t+1@.
+One important aspect of block RAMs is that they have a /synchronous/ read port,
+meaning that, unlike the behavior of 'Clash.Prelude.RAM.asyncRam', given a read
+address @r@ at time @t@, the value @v@ in the RAM at address @r@ is only
+available at time @t+1@.
 
 For us that means we need to change the design of our CPU. Right now, upon a
 load instruction we generate a read address for the memory, and the value at
 that read address is immediately available to be put in the register bank.
-Because we will be using a BlockRAM, the value is delayed until the next cycle.
-We hence need to also delay the register address to which the memory address
-is loaded:
+Because we will be using a block RAM, the value is delayed until the next cycle.
```

Comment:
This isn't really a correct way to start a sentence with "Because". Generally speaking I would avoid it, and instead write something like

> We will be using a block RAM, so the value is delayed until the next cycle. Thus, we will need to also delay the register address to which the memory address is loaded:

A more proper way still starting with "Because" would be

> Because values are delayed until the next cycle when using a block RAM, we need to delay the register address to which the memory address is loaded:

Reply:
I also moved the _also_ in the last sentence:

> Thus, we will also need to delay the register address to which the memory address is loaded:

Do you agree that's a more natural word order?

Or would you rather I say "I moved the _also_ also in the last sentence"? 😜

Reply:
That's _also_ fine :wink: